EDDiscovery/Trilateration/TrilaterationControl.cs

```diff
@@ -80,17 +84,17 @@ private void dataGridViewDistances_EditingControlShowing(object sender, DataGrid
             {
                 textbox.AutoCompleteMode = AutoCompleteMode.None;
                 return;
-            }
-            
+            }
+
             textbox.AutoCompleteMode = AutoCompleteMode.SuggestAppend;
             textbox.AutoCompleteSource = AutoCompleteSource.CustomSource;
             var items = new AutoCompleteStringCollection();
-
+            
             var enteredSystems = GetEnteredSystems();
             items.AddRange((
                 from s
                 in SystemData.SystemList
-                where s.HasCoordinate && (s.name == textbox.Text || !enteredSystems.Contains(s))
+                where s.HasCoordinate && (s.name == textbox.Text || enteredSystems.Where(lu => lu.name == s.name).Count() == 0)
```

Comment:
I'm not entirely sure why this works when the .Contains(s) didn't but it does (on my box anyway....)